alt_e2eshark/onnx_tests/helper_classes.py

```diff
@@ -23,6 +25,50 @@
 """This file contains several helpful child classes of OnnxModelInfo."""
 
 
+class HfDownloadableModel(OnnxModelInfo):
+    """This class should be used to download models from Huggingface model hub."""
+
+    def __init__(self, full_model_path, task_name, name, onnx_model_path):
+        opset_version = 21
+        self.model_repo_path = full_model_path.replace("hf_", "")
+        self.task = task_name
+        super().__init__(name, onnx_model_path, opset_version)
+
+    def export_model(self):
+        model_dir = str(Path(self.model).parent)
+        # TODO: Use Python API instead of CLI?
+        opt_cli_command = f"optimum-cli export onnx --model {self.model_repo_path} --task {self.task} --monolith --framework pt {model_dir}"
```

Comment:
IIRC, optimum doesn't support exporting some model types like FLUX. 

I'm also wondering if we should avoid using OnnxModelInfo, and instead define a new ModelInfo base class for models originally in pytorch, which we could run natively in pytorch for the gold output. 

The reason I'd probably prefer this approach is that for the large models we eventually want to export with initializers converted to inputs, we won't really be able to run those with ORT. 

Let me think about this a bit more. It might be fine to leave this as-is for the medium sized models and do something else for the extremely large models.

Reply:
That would be easy to do. I wrote it this way so that we can have a custom registry for very large models or the ones that optimum doesn't support, and override this function for those few models. Having optimum for the bigger set allows for easier implementation, and debugging IMO.